airflow/providers/amazon/aws/operators/glue.py

```diff
@@ -239,3 +243,253 @@ def on_kill(self):
+        (default: False)
+
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+        If this is ``None`` or empty then the default boto3 behaviour is used. If
+        running Airflow in a distributed manner and aws_conn_id is None or
+        empty, then default boto3 configuration would be used (and must be
+        maintained on each worker node).
+    :param region_name: AWS region_name. If not specified then the default boto3 behaviour is used.
+    :param verify: Whether or not to verify SSL certificates. See:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html
+    :param botocore_config: Configuration dictionary (key-values) for botocore client. See:
+        https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html
+    """
+
+    aws_hook_class = GlueDataQualityHook
+
+    template_fields: Sequence[str] = (
+        "datasource",
+        "role",
+        "rule_set_names",
+        "rule_set_evaluation_run_kwargs",
+    )
+
+    template_fields_renderers = {"datasource": "json", "rule_set_evaluation_run_kwargs": "json"}
+
+    ui_color = "#ededed"
+
+    def __init__(
+        self,
+        *,
+        datasource: dict,
+        role: str,
+        rule_set_names: list[str],
+        number_of_workers: int = 5,
+        timeout: int = 2880,
+        verify_result_status: bool = True,
+        show_results: bool = True,
+        rule_set_evaluation_run_kwargs: dict[str, Any] | None = None,
+        wait_for_completion: bool = True,
+        waiter_delay: int = 60,
+        waiter_max_attempts: int = 20,
+        deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
+        aws_conn_id: str | None = "aws_default",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.datasource = datasource
+        self.role = role
+        self.rule_set_names = rule_set_names
+        self.number_of_workers = number_of_workers
+        self.timeout = timeout
+        self.verify_result_status = verify_result_status
+        self.show_results = show_results
+        self.rule_set_evaluation_run_kwargs = rule_set_evaluation_run_kwargs or {}
+        self.wait_for_completion = wait_for_completion
+        self.waiter_delay = waiter_delay
+        self.waiter_max_attempts = waiter_max_attempts
+        self.deferrable = deferrable
+        self.aws_conn_id = aws_conn_id
+
+    def validate_inputs(self) -> None:
+        glue_table = self.datasource.get("GlueTable", {})
+
+        if not glue_table.get("DatabaseName") or not glue_table.get("TableName"):
+            raise AttributeError("DataSource glue table must have DatabaseName and TableName")
+
+        not_found_ruleset = []
+
+        for ruleset_name in self.rule_set_names:
+            if not self.hook.has_data_quality_ruleset(ruleset_name):
+                not_found_ruleset.append(ruleset_name)
```

Comment:
```suggestion
        not_found_ruleset = [ruleset_name for ruleset_name in self.rule_set_names if not self.hook.has_data_quality_ruleset(ruleset_name)]
```

Reply:
Thank you @vincbeck , Appreciate your feedback 😄 . I will update changes.

Reply:
updated the changes.